### PR TITLE
feat(w3c): remove processVersion option

### PR DESCRIPTION
**Pull request by marcoscaceres**
* [ ] Update Wiki to reflect this change once this is merged.

### CODE REVIEW

**Comment by saschanaz** [comment]
Not sure why it should be string, will it be semver later?

**Comment by marcoscaceres** [comment]
There are some equality checks (though they rely on coercion via "=="):
https://github.com/w3c/respec/blob/develop/src/w3c/headers.js#L654

I want to make these strict checks, and always just use a single data type.

**Comment by marcoscaceres** [comment]
Maybe I'm being overly picky 🙄

**Comment by saschanaz** [comment]
Hmm. It seems processVersion is always `20xx`, can we make it a number and fix tests that thinks it as a string?

**Comment by marcoscaceres** [comment]
Yeah, that's a good suggestion. Will switch this around.

File: src/w3c/headers.js [hunk 1]
```diff
@@ -651,14 +653,27 @@ export function run(conf, doc, cb) {
       return urlParts[pos] || "";
     },
   });
-  if (conf.processVersion == "2014" || conf.processVersion == "2015" || conf.processVersion == "2017") {
-    pub(
-      "warn",
-      "Process " + conf.processVersion + " has been superceded by Process 2018."
-    );
-    conf.processVersion = "2018";
+  if (W3CProcessVersion.known.includes(conf.processVersion) === false) {
```
Comment (saschanaz) [inline src/w3c/headers.js]: nit: `!x` rather than `x === false`, as `includes` always returns Boolean.

File: src/w3c/headers.js [hunk 2]
```diff
@@ -651,14 +653,27 @@ export function run(conf, doc, cb) {
...
+    const msg =
+      `Unknown W3C Process Document version: "${
+        conf.processVersion
+      }". Please update your ` +
```
Comment (saschanaz) [inline src/w3c/headers.js]: Can this be in a single line: `"${conf.processVersion}"`?

File: src/w3c/headers.js [hunk 3]
```diff
@@ -651,14 +653,27 @@ export function run(conf, doc, cb) {
...
+      `to one of: ${W3CProcessVersion.known.join(", ")}.`;
+    pub("error", msg);
+    conf.processVersion = W3CProcessVersion.current;
+  } else if (conf.processVersion < W3CProcessVersion.current) {
+    const msg =
+      `The W3C's ${
+        conf.processVersion
+      } Process Document has been superceded by the ${
+        W3CProcessVersion.current
+      } Process Document. Please update, or remove, your ` +
```
Comment (saschanaz) [inline src/w3c/headers.js]: (Same here)

**Comment by marcoscaceres** [comment]
@plehegar or @dontcallmedom, is there any reason for any document to still be using the 2005 (!) process document version? If, no, then I can just force all documents to the 2018 process and get rid of all this code.

File: src/w3c/headers.js (hunk 2 above)
Comment (marcoscaceres) [reply src/w3c/headers.js]: Prettier.js keeps formatting it like that... "computer says no" 😂

Comment (saschanaz) [reply src/w3c/headers.js]: Line 662 is okay but line 660 is not okay? That's weird 🤣

Comment (marcoscaceres) [reply src/w3c/headers.js]: I'm still hopeful we can delete all things (best kind of code). I have good feeling @plehegar will say "yes".

**Comment by marcoscaceres** [comment]
In anticipation of an ok from @plehegar, I've preemptively updated this so it removes the configurable `processVersion` member entirely. 

If @plehegar decides for whatever reason we want to keep it,  we can just revert my last commit.

**Comment by plehegar** [comment]
@marcoscaceres no reason to keep process 2005 around imho, so +1 from me.

**Comment by marcoscaceres** [comment]
@saschanaz, given ok we got from @plehegar, could you review again?

**Pull request merged**